src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java

```diff
@@ -79,6 +87,58 @@
 
     private static final int DEFAULT_SLAVE_TIMEOUT = 900;
 
+    /**
+     * Start auto scaling ECS clusters as part of Jenkins initialization.
+     *
+     * @throws InterruptedException
+     * @throws IOException
+     */
+
+    @Initializer(after = InitMilestone.JOB_LOADED)
+    public static void init() throws InterruptedException, IOException {
+        final Jenkins jenkins = Jenkins.getInstance();
+
+        // Remove all slaves that were persisted when Jenkins shutdown.
+        for (final ECSSlave ecsSlave : getSlaves()) {
+            LOGGER.log(Level.INFO, "Terminating ECS slave {0}", new Object[] {ecsSlave.getDisplayName()});
+            ecsSlave.terminate();
+        }
+
+        // Start auto scale in
+        for (final Cloud c : jenkins.clouds) {
+            if (c instanceof ECSCloud) {
+                final ECSCloud ecsCloud = (ECSCloud)c;
+                ecsCloud.startAutoScaleIn();
+            }
+        }
+    }
+
+    public static List<ECSSlave> getSlaves() {
+        final Jenkins jenkins = Jenkins.getInstance();
+        final List<ECSSlave> ecsSlaves = new ArrayList<ECSSlave>();
+        final List<Node> allSlaves = jenkins.getNodes();
+        for (final Node n : allSlaves) {
+            if (n instanceof ECSSlave) {
+                final ECSSlave ecsSlave = (ECSSlave)n;
```

Comment:
Hello :wave:, random reader here.
It seems you forgot to append the `ECSSlave` to the `ecsSlaves` result list.